cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -243,22 +243,22 @@ type Networking struct {
 type BootstrapToken struct {
 	// Token is used for establishing bidirectional trust between nodes and control-planes.
 	// Used for joining nodes in the cluster.
-	Token *BootstrapTokenString
+	Token *BootstrapTokenString `json:"token"`
```

Comment:
@bart0sh `apis/kubeadm/types.go` so we cannot add `json` fields to it...it's not supposed to be marshaled.
buit in `apis/kubeadm/v1beta1/types.go` there is a public type that already has these.

Reply:
Thank you for pointing that out to me. I'm a bit confused. All helpers in bootstraptokenhelpers.py (BootstrapTokenFromSecret in this particular case) work with kubeadmapi.BootstrapToken. Should I copy results to  kubeadmapiv1beta1.BootstrapToken struct or there is a better way?

Reply:
re: `bootstraptokenhelpers.py` do you mean `bootstraptokenhelpers.go`?
if the helpers only work with the private API (the one with `json` tags`) one way to deal with the problem is to convert the private type to public before marshaling. i will leave this to @rosti to comment further.

Reply:
@neolit123 yes, I meant bootstraptokenhelpers.go. Sorry for the confusion.
@rosti I'm still confused. Is copying data from private structure to identical public structure just for marshalling a common practice in kubeadm? Can you point me out to the kubeadm code that does that, please? There should be better way to handle this in my opinion.

Reply:
@rosti I'd rather change BootstrapTokenFromSecret and other helpers to work with public structures. That would be more logical in my opinion.